go/enclave/events/subscription_manager.go

```diff
@@ -97,12 +100,25 @@ func (s *SubscriptionManager) EncryptLogs(logsBySubID map[uuid.UUID][]*types.Log
 	return result, nil
 }
 
-// Indicates whether the log is relevant for any subscription, and returns the matching subscription account if so.
-// A lifecycle log is considered relevant to everyone.
-// TODO - #453 - Filter logs, instead of considering all logs relevant to everyone.
-func isRelevant(log *types.Log, sub *common.LogSubscription) (bool, *common.SubscriptionAccount) {
-	// Work out if this is an account or lifecycle log
-	// If the former, establish whether it is relevant to any subscription
-	// Return the first account for which the log matches, so it can be used for encryption
-	return true, nil
+// Indicates whether the log is relevant for the subscription. A lifecycle log is considered relevant to everyone.
+func isRelevant(log *types.Log, sub *common.LogSubscription, db *state.StateDB) bool {
+	var nonContractAddrs []string
+	for _, topic := range log.Topics {
+		addr := gethcommon.HexToAddress(topic.Hex())
```

Comment:
it would make sense to filter out topics that are not exactly 20 bytes

Reply:
Good idea, thanks